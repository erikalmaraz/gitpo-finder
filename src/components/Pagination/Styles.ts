import styled from "styled-components";

export const PaginationContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 20px 0;
  align-items: center;
`;

export const PaginationMessage = styled.div`
  font-size: 0.9rem;
  font-weight: bold;
`;

export const PaginationControllers = styled.div`
  display: flex;
`;

export const ButtonControllers = styled.button`
  background: ${({ theme }) => (theme === "light" ? "#333" : "#f9f9f9")};
  color: ${({ theme }) => (theme === "light" ? "#fff" : "#333")};
  border: 0;
  padding: 8px;
  border-radius: 6px;
  margin: 0 8px;
`;

export const PositionIndicator = styled.button`
  width: 30px;
  text-align: center;
  border: 1px solid #ddd;
  border-radius: 4px;
  padding: 6px;
  background-color: #eee;
  margin: 0 2px;
`;
