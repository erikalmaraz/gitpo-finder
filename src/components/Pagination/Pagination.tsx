import React from "react";
import PaginationType from "../../interfaces/pagination.interface";
import * as S from "./Styles";
const Pagination = ({ totalItems, nextPage, prevPage }: PaginationType) => {
  return (
    <React.Fragment>
      <S.PaginationContainer>
        <S.PaginationMessage>We found {totalItems} repos.</S.PaginationMessage>
        <S.PaginationControllers>
          <S.ButtonControllers onClick={prevPage}>Prev</S.ButtonControllers>
          <S.ButtonControllers onClick={nextPage}>Next</S.ButtonControllers>
        </S.PaginationControllers>
      </S.PaginationContainer>
    </React.Fragment>
  );
};

export default Pagination;
