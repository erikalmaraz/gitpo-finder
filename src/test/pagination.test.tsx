import React from "react";
import { render, screen } from "@testing-library/react";
import Pagination from "../components/Pagination/Pagination";
import renderer from "react-test-renderer";

it("Should render Pagination correctly", () => {
  const props = {
    totalItems: 2,
    nextPage: jest.fn,
    prevPage: jest.fn,
  };
  const tree = renderer.create(<Pagination {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
