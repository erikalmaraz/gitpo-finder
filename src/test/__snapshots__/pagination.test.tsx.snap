// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Pagination correctly 1`] = `
<div
  className="sc-bdfBQB iUljAX"
>
  <div
    className="sc-gsTEea jmguBT"
  >
    We found 
    2
     repos.
  </div>
  <div
    className="sc-dlfnuX eFAgps"
  >
    <button
      className="sc-hKgJUU kWBnnj"
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="sc-hKgJUU kWBnnj"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
